src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```
@@ -0,0 +1,61 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import mate.academy.springboot.datajpa.domain.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.repository.specification.ProductSpecificationManager;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+    private final ProductSpecificationManager productSpecificationManager;
+
+    public ProductServiceImpl(ProductRepository productRepository,
+                              ProductSpecificationManager productSpecificationManager) {
+        this.productRepository = productRepository;
+        this.productSpecificationManager = productSpecificationManager;
+    }
+
+    @Override
+    public Product create(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product findById(Long id) {
+        return productRepository.findById(id).orElseThrow();
```

let's throw an exception with a custom message